### PR TITLE
[booster] update bert example, using booster api

### PR DESCRIPTION
**Pull request by wukong1992**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> update bert example using booster api
> if you have any plots/diagrams/screenshots/tables, please attach them here.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: examples/benchmark/bert.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import argparse
```
Comment (ver217) [inline examples/benchmark/bert.py:1]: Place this file under `language/bert`

File: examples/README.md [hunk 2]
```diff
@@ -56,6 +57,13 @@ Therefore, it is essential for the example contributors to know how to integrate
 3. Export your dataset path with the prefix `/data` and make sure you have a copy of the dataset in the `/data/scratch/examples-data` directory on the CI machine. Community contributors can contact us via slack to request for downloading the dataset on the CI machine.
 4. Implement the logic such as dependency setup and example execution
 
+## Benchmark Your Example
```
Comment (ver217) [inline examples/README.md:60]: This content is outdated and should be moved to `language/bert/README.md`

File: examples/language/bert/bench_mark.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -xe
```
Comment (ver217) [inline examples/language/bert/bench_mark.sh:2]: Please rename this file to `benchmark.sh`

File: examples/language/bert/benchmark.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
+import inspect
```
Comment (ver217) [inline examples/language/bert/benchmark.py:1]: Renaming this file to `benchmark_utils.py` would be better.

File: examples/language/bert/bert.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
+import argparse
```
Comment (ver217) [inline examples/language/bert/bert.py:1]: Renaming this file to `benchmark.py` would be better.

File: examples/language/bert/bert.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+        plugin = GeminiPlugin(placement_policy='cuda', strict_ddp_mode=True, initial_scale=2**5)
+    elif args.plugin == 'low_level_zero':
+        plugin = LowLevelZeroPlugin(initial_scale=2**5)
+
+    booster = Booster(plugin=plugin, **booster_kwargs)
+
+    # ==============================
+    # Prepare Dataloader
+    # ==============================
+    data_builder = GLUEDataBuilder(model_name,
```
Comment (ver217) [inline examples/language/bert/bert.py:90]: For sake of simplicity, we'd better use systhetic random dataset here.

File: examples/language/bert/bert.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    train_dataloader = data_builder.train_dataloader()
+
+    # ====================================
+    # Prepare model, optimizer
+    # ====================================
+    # bert pretrained model
+
+    cfg = AutoConfig.from_pretrained(model_name, num_labels=data_builder.num_labels)
+    if model_name == "bert-base-uncased":
+        model = BertForSequenceClassification.from_pretrained(model_name, config=cfg)
```
Comment (ver217) [inline examples/language/bert/bert.py:104]: Initializing from pretrained will download weights, which cost much time. We'd better initialize from config (scratch).

File: examples/language/bert/bert.py [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+    results = benchmark(model,
+                        booster,
+                        optimizer,
+                        lr_scheduler,
+                        train_dataloader,
+                        criterion=criterion,
+                        epoch_num=NUM_EPOCHS)
+
+    if coordinator.is_master():
```
Comment (ver217) [inline examples/language/bert/bert.py:154]: `coordinator.print_on_master()` may be simpler.

File: examples/README.md (hunk 2 above)
Comment (ver217) [reply examples/README.md:60]: Please add a readme under `language/bert` including `setup`, `usage` and `result` for both finetuning example and benchmark.

File: examples/language/bert/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+Now include these metrics in benchmark: CUDA mem occupy，throughput and the number of model parameters. If you have a custom metrics, you can add it to benchmark_util.
+
+## Results
+
+### bert
+
+|       | max cuda mem | throughput(sample/s) | params |
+| :-----| -----------: | :--------: | :----: |
+| ddp | 23.02 G | 0.5 | 82M|
+| ddp_fp16 | 17.46 G | 5 |82M|
```
Comment (ver217) [inline examples/language/bert/README.md:24]: Please add space around `82M`, etc.

File: examples/language/bert/README.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+```
+bash test_ci.sh
+```
+
+## Benchmark
+```
+bash benchmark.sh
+```
+
+Now include these metrics in benchmark: CUDA mem occupy，throughput and the number of model parameters. If you have a custom metrics, you can add it to benchmark_util.
```
Comment (ver217) [inline examples/language/bert/README.md:15]: It seems that the first comma is the comma in Chinese. And this sentence has syntax error. You can use some tools like https://www.nounplus.net/grammarcheck/ to check your sentence.

File: examples/language/bert/README.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+## Results
+
+### bert
```
Comment (ver217) [inline examples/language/bert/README.md:19]: Please use camel case

File: examples/language/bert/benchmark.py [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+        self._vocab_size = vocab_size
+        self._n_class = n_class
+        self._device = device
+        self._dataset_length = dataset_length
+
+    def __len__(self):
+        return self._dataset_length
+
+    def __getitem__(self, idx):
+        input = torch.randint(
```
Comment (ver217) [inline examples/language/bert/benchmark.py:50]: Initializing a big tensor in `__init__()` and slicing it here would be faster.

**Review by ver217 — APPROVED** [review]

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**